central/resourcecollection/service/service_impl.go

```diff
@@ -242,21 +233,38 @@ func collectionRequestToCollection(ctx context.Context, request collectionReques
 	return collection, nil
 }
 
+func resolveQuery(rawQuery *v1.RawQuery, withPagination bool) (*v1.Query, error) {
+	if rawQuery == nil {
+		query := search.EmptyQuery()
+		if withPagination {
+			paginated.FillPagination(query, &v1.Pagination{}, defaultPageSize)
+			return query, nil
+		}
+
+		return search.EmptyQuery(), nil
+	}
+
+	query, err := search.ParseQuery(rawQuery.GetQuery(), search.MatchAllIfEmpty())
+	if err != nil {
+		return nil, err
+	}
+	if withPagination {
+		paginated.FillPagination(query, rawQuery.GetPagination(), defaultPageSize)
+	}
+	return query, nil
```

Comment:
```suggestion
	query, err := search.ParseQuery(rawQuery.GetQuery(), search.MatchAllIfEmpty())
	if err != nil {
		return nil, err
	}
	if withPagination {
		paginated.FillPagination(query, rawQuery.GetPagination(), defaultPageSize)
	}
	return query, nil
```
- `search.ParseQuery` with the match all option ensures that `query` is set to `EmptyQuery()` along with nil check.

Reply:
`rawQuery` could be nil so we don't want to just reference into it, that's what was causing the panics that David was encountering when passing it as `query` instead of `query.query`. That's the reason this function looks so unruly, it's because I'm trying to cover that case

Reply:
<img width="612" alt="image" src="https://user-images.githubusercontent.com/9895898/203157422-242b9fd1-9e5b-4c57-b0aa-89df21ff5c83.png">